src/adapters/vertexai/chat.ts

```diff
@@ -0,0 +1,165 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isString } from "remeda";
+import {
+  AsyncStream,
+  BaseLLMTokenizeOutput,
+  ExecutionOptions,
+  GenerateCallbacks,
+  GenerateOptions,
+  LLMCache,
+  LLMMeta,
+} from "@/llms/base.js";
+import { shallowCopy } from "@/serializer/utils.js";
+import type { GetRunContext } from "@/context.js";
+import { Emitter } from "@/emitter/emitter.js";
+import { VertexAI, GenerativeModel, ModelParams } from "@google-cloud/vertexai";
+import { ChatLLM, ChatLLMOutput } from "@/llms/chat.js";
+import { BaseMessage, Role } from "@/llms/primitives/message.js";
+import { signalRace } from "@/internals/helpers/promise.js";
+import { processContentResponse, registerGenerativeModel } from "./utils.js";
+
+export class VertexAIChatLLMOutput extends ChatLLMOutput {
+  public readonly chunks: BaseMessage[] = [];
+
+  constructor(chunk: BaseMessage) {
+    super();
+    this.chunks.push(chunk);
+  }
+
+  get messages() {
+    return this.chunks;
+  }
+
+  merge(other: VertexAIChatLLMOutput): void {
+    this.chunks.push(...other.chunks);
+  }
+
+  getTextContent(): string {
+    return this.chunks.map((result) => result.text).join("");
+  }
+
+  toString() {
+    return this.getTextContent();
+  }
+
+  createSnapshot() {
+    return { chunks: shallowCopy(this.chunks) };
+  }
+
+  loadSnapshot(snapshot: typeof this.createSnapshot): void {
+    Object.assign(this, snapshot);
+  }
+}
+
+export interface VertexAIChatLLMInput extends GenerateOptions {
+  modelId: string;
+  project: string;
+  location: string;
+  client?: GenerativeModel;
+  executionOptions?: ExecutionOptions;
+  cache?: LLMCache<VertexAIChatLLMOutput>;
+  parameters?: Record<string, any>;
+}
+
+export class VertexAIChatLLM extends ChatLLM<VertexAIChatLLMOutput> {
+  public readonly emitter: Emitter<GenerateCallbacks> = Emitter.root.child({
+    namespace: ["vertexai", "llm"],
+    creator: this,
+  });
+
+  protected client: GenerativeModel;
+
+  constructor(protected readonly input: VertexAIChatLLMInput) {
+    super(input.modelId, input.executionOptions, input.cache);
+    const modelParams: ModelParams = { model: input.modelId };
+    const schema = isString(input.guided?.json)
+      ? JSON.parse(input.guided?.json)
+      : input.guided?.json;
+    const generationConfig = { responseSchema: schema, responseMimeType: "application/json" };
+    if (schema !== null && schema !== undefined) {
+      modelParams.generationConfig = generationConfig;
+    }
```

Comment:
Guided can be passed via `GenerateOptions` in run options. You should create a standalone method that merges the incoming `options` (GenerateOptions) with class-defined `parameters` (as done in the BAM adapter in the `_prepareParameters` method).